src/Symfony/Component/DomCrawler/Tests/CrawlerTest.php

```diff
@@ -382,16 +382,35 @@ public function testNodeName()
         }
     }
 
+    /**
+     * @group legacy
+     * @expectedException \InvalidArgumentException
+     */
     public function testText()
     {
-        $this->assertEquals('One', $this->createTestCrawler()->filterXPath('//li')->text(), '->text() returns the node value of the first element of the node list');
+        $this->assertEquals("one   two\nthree   ", $this->createTestCrawler()->filterXPath('//div[@id="text-whitespaces"]')->text(), '->text() returns the node value of the first element of the node list');
 
-        try {
-            $this->createTestCrawler()->filterXPath('//ol')->text();
-            $this->fail('->text() throws an \InvalidArgumentException if the node list is empty');
-        } catch (\InvalidArgumentException $e) {
-            $this->assertTrue(true, '->text() throws an \InvalidArgumentException if the node list is empty');
-        }
+        $this->createTestCrawler()->filterXPath('//ol')->text();
+    }
+
+    /**
+     * @expectedException \InvalidArgumentException
+     */
+    public function testTextWithNormalizedWhitespaces()
+    {
+        $this->assertEquals('one two three', $this->createTestCrawler()->filterXPath('//div[@id="text-whitespaces"]')->text(true), '->text() returns the node value of the first element of the node list');
+
+        $this->createTestCrawler()->filterXPath('//ol')->text(true);
+    }
+
+    /**
+     * @expectedException \InvalidArgumentException
+     */
+    public function testValue()
+    {
+        $this->assertEquals("one   two\nthree   ", $this->createTestCrawler()->filterXPath('//div[@id="text-whitespaces"]')->value(), '->value() returns the node value of the first element of the node list');
+
+        $this->createTestCrawler()->filterXPath('//ol')->value();
```

Comment:
this should be a separate test like `testValueThrowsExceptionWhenNodeListEmpty`. if the above call aleady fails with InvalidArgumentException, the test would still pass.